hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/client/SparkRDDMetadataWriteClient.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.hudi.client;
+
+import org.apache.hudi.common.engine.HoodieEngineContext;
+import org.apache.hudi.config.HoodieWriteConfig;
+
+/**
+ * Metadata write client for spark. 
+ * @param <T>
+ */
+public class SparkRDDMetadataWriteClient<T> extends SparkRDDWriteClient<T> {
+  public SparkRDDMetadataWriteClient(HoodieEngineContext context, HoodieWriteConfig clientConfig) {
+    super(context, clientConfig);
+  }
+
+  /**
+   * In Metadata table, we may need to rollback a commit(partially failed) which may not be the latest in commit time ordering.
+   * since commit time in metadata timeline is just a replica of data table commit times, to rollback any partially failed commits, we have to expose this api
+   * just for metadata table.
+   */
+  public void rollbackPartiallyFailedWrites() {
+    this.rollbackFailedWrites();
```

Comment:
nit: Shall we make `rollbackFailedWrites()` public and avoid adding this class (`SparkRDDMetadataWriteClient`) all together?

Reply:
@xushiyan @alexeykudinkin : any thoughts here.

Reply:
Yes, we should avoid creating bespoke components specific to MT